src/lib/devices/DeviceCard/DeviceRoutes.svelte

```diff
@@ -22,7 +22,17 @@
 	}
 
   function enableDeviceRouteAction(route: string) {
-    enableDeviceRoute(device.id, route)
+    enableDeviceRoute(device.id, [route,...routesList.enabledRoutes])
+			.then((response) => {
+				getDeviceRoutesAction();
+			})
+			.catch((error) => {
+				$alertStore = error;
+			});
+  }
+
+  function disableDeviceRouteAction(route: string) {
+    enableDeviceRoute(device.id, [...routesList.enabledRoutes].filter(v=>v!=route))
```

Comment:
What's `v=>v!=route` doing in this situation? Can you add a comment?

Reply:
The function `disableDeviceRouteAction ` needs to make the route selected by the user(`route`) to be inactive, So we need to pass the active routing list to method `enableDeviceRoute` excluding `route` by using `filter(v=>v!=route)`.
And `enableDeviceRoute(device.id, [...routesList.enabledRoutes].filter(v=>v!=route))` is same to
`var routes = [];
for(let v of routesList){
    if(v != route){
        routes.push(v)
    }
}
enableDeviceRoute(device.id, routes)
`